File: src/main/java/com/telegram/folobot/service/FoloUserService.java
package com.telegram.folobot.service;

import com.telegram.folobot.domain.FoloUserEntity;
import com.telegram.folobot.dto.FoloUserDto;
import com.telegram.folobot.mappers.FoloUserMapper;
import com.telegram.folobot.repos.FoloUserRepo;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
@RequiredArgsConstructor
public class FoloUserService {
    private final FoloUserRepo foloUserRepo;
    private final FoloUserMapper foloUserMapper;

    /**
     * Прочитать все
     * @return {@link List<FoloUserDto>}
     */
    public List<FoloUserDto> findAll() {
        return foloUserRepo.findAll().stream()
                .map(foloUserMapper::mapToFoloUserDto)
                .toList();
    }

    /**
     * Чтение по Id
     * @param userId {@link Long} Id
     * @return {@link FoloUserDto}
     */
    public FoloUserDto findById(Long userId) {
        return foloUserMapper.mapToFoloUserDto(foloUserRepo.findById(userId)
                        .orElse(new FoloUserEntity(userId)));
    }

    /**
     * Проверка наличия
     * @param userId Id пользователя
     * @return да/нет
     */
    public boolean existsById(Long userId) {
        return foloUserRepo.existsById(userId);
    }

    /**
     * Сохранение
     * @param dto {@link FoloUserDto}
     */
    public void save(FoloUserDto dto) {
        foloUserRepo.save(foloUserMapper.mapToFoloUserEntity(dto));
    }

    /**
     * Удаление
     * @param dto {@link FoloUserDto}
     */
    public void delete(FoloUserDto dto) {foloUserRepo.delete(foloUserMapper.mapToFoloUserEntity(dto)); }
}

File: src/main/kotlin/com/telegram/folobot/service/FoloUserService.kt
package com.telegram.folobot.service

import com.telegram.folobot.persistence.dto.FoloUserDto
import com.telegram.folobot.persistence.dto.toEntity
import com.telegram.folobot.persistence.entity.FoloUserEntity
import com.telegram.folobot.persistence.entity.toDto
import com.telegram.folobot.persistence.repos.FoloUserRepo
import org.springframework.stereotype.Component

@Component
class FoloUserService(private val foloUserRepo: FoloUserRepo) {
    /**
     * Прочитать все
     * @return [<]
     */
    fun findAll(): List<FoloUserDto> {
        return foloUserRepo.findAll().map { it.toDto() }
    }

    /**
     * Чтение по Id
     * @param userId [Long] Id
     * @return [FoloUserDto]
     */
    fun findById(userId: Long): FoloUserDto {
        return foloUserRepo.findUserByUserId(userId)?.toDto() ?: FoloUserEntity(userId).toDto()
    }

    /**
     * Проверка наличия
     * @param userId Id пользователя
     * @return да/нет
     */
    fun existsById(userId: Long): Boolean {
        return foloUserRepo.existsById(userId)
    }

    /**
     * Сохранение
     * @param dto [FoloUserDto]
     */
    fun save(dto: FoloUserDto) {
        foloUserRepo.save(dto.toEntity())
    }

    /**
     * Удаление
     * @param dto [FoloUserDto]
     */
    fun delete(dto: FoloUserDto) {
        foloUserRepo.delete(dto.toEntity())
    }
}